src/gt4py/frontend/gtscript_frontend.py

```diff
@@ -1163,6 +1172,12 @@ def visit_Assign(self, node: ast.Assign) -> list:
                     else:
                         result.append(field_decl)
                     self.fields[field_decl.name] = field_decl
+                else:
+                    if len(self.fields[t.id].axes) == 1:
+                        raise GTScriptSyntaxError(
```

Comment:
This should also be raised in the `isinstance(t, ast.Subscript)` case.

Reply:
Also, it should also not be possible to write to _any_ reduced dimensionality field, with the exception of IJ fields in non-parallel regions.

Reply:
Good catch. I added this in general for the target of the assignment, and made the check that all parallel axes are present (including K if in a parallel computation).